token/program-2022-test/tests/reallocate.rs

```diff
@@ -188,3 +188,52 @@ async fn reallocate_without_current_extension_knowledge() {
         ])
     );
 }
+
+#[tokio::test]
+async fn reallocate_syncs_native() {
+    let mut context = TestContext::new().await;
+    context.init_token_with_native_mint().await.unwrap();
+    let TokenContext { token, alice, .. } = context.token_context.unwrap();
+    let context = context.context.clone();
+
+    let alice_account = Keypair::new();
+    token
+        .create_auxiliary_token_account(&alice_account, &alice.pubkey())
+        .await
+        .unwrap();
+    let alice_account = alice_account.pubkey();
+
+    let token_account = token.get_account_info(&alice_account).await.unwrap();
+    let pre_amount = token_account.base.amount;
+    let pre_is_native = token_account.base.is_native.unwrap();
```

Comment:
```suggestion
    let pre_rent_exempt_reserve = token_account.base.is_native.unwrap();
```

Reply:
(Sorry, several of these are just nits to make the test more readable. Take 'em or leave 'em)